README.md

```diff
@@ -46,102 +64,152 @@ Whether to use SSL/TLS for the connection. Defaults to `true`.
 
 Timeout value in seconds. Must be between 1 and 60 seconds.
 
+
 ## Examples
 
-Show how your plugin is to be used
+### Basic usage
+
+Minimal configuration with just the required option:
 
 ```yaml
 steps:
-  - label: "🔨 Running plugin"
-    command: "echo template plugin"
+  - label: "🔨 Basic plugin usage"
+    command: "echo processing"
     plugins:
       - template#v1.0.0:
-          mandatory: "value"
+          mandatory: "required-value"
 ```
 
-## And with other options as well
+### With optional parameters
 
-If you want to change the plugin behaviour:
+Adding optional configuration:
 
 ```yaml
 steps:
-  - label: "🔨 Running plugin"
-    command: "echo template plugin with options"
+  - label: "🔨 Plugin with options"
+    command: "echo processing with options"
     plugins:
       - template#v1.0.0:
-          mandatory: "value"
-          optional: "example"
+          mandatory: "required-value"
+          optional: "custom-value"
+          timeout: 45
 ```
 
+### Array processing
+
+Handling arrays of values:
+
 ```yaml
 steps:
   - label: "🔨 Array processing"
-    command: "echo processing array"
+    command: "echo processing numbers"
     plugins:
       - template#v1.0.0:
-          mandatory: "value"
+          mandatory: "required-value"
           numbers: [1, 2, 3, 5, 8]
 ```
 
+### Feature toggles
+
+Using boolean flags to control behavior:
+
 ```yaml
 steps:
-  - label: "🔨 Feature toggle"
-    command: "echo feature processing"
+  - label: "🔨 Feature enabled"
+    command: "echo enhanced processing"
     plugins:
       - template#v1.0.0:
-          mandatory: "value"
+          mandatory: "required-value"
           enabled: true
 ```
 
+### Complex configuration
+
+Using nested configuration objects:
+
 ```yaml
 steps:
-  - label: "🔨 Configuration"
-    command: "echo processing config"
+  - label: "🔨 Complex config"
+    command: "echo connecting to service"
     plugins:
       - template#v1.0.0:
-          mandatory: "value"
+          mandatory: "required-value"
           config:
-            host: "example.com"
+            host: "api.example.com"
             port: 8080
             ssl: false
 ```
 
+### Secrets and environment variables
+
+Secure handling of secrets using environment variables:
+
+```yaml
+steps:
+  - label: "🔨 Using secrets"
+    command: "echo authenticated processing"
+    plugins:
+      - template#v1.0.0:
+          mandatory: "required-value"
+          optional: "MY_SECRET_TOKEN"  # Pass env var name instead of secret value
+    env:
+      MY_SECRET_TOKEN: "secret-value-here"
+```
+
+In the plugin code, use `${!config_value}` to get the secret value from the environment variable name.
```

Comment:
suggestion for the future: a helper function that you pass a plugin option but dereferences it this way

Reply:
Created https://github.com/buildkite-plugins/template-buildkite-plugin/issues/17 to handle it later. Thanks!